pkg/component/shared/istio.go

```diff
@@ -239,6 +259,37 @@ func IsZonalIstioExtension(labels map[string]string) (bool, string) {
 	return false, ""
 }
 
+// ShouldEnforceSpreadAcrossHosts checks whether all given zones have at least two nodes so that Istio can be spread across hosts in each zone
+func ShouldEnforceSpreadAcrossHosts(ctx context.Context, cl client.Client, zones []string) (bool, error) {
+	const targetNodeCount = 2
+	nodeList := &metav1.PartialObjectMetadataList{}
+	nodeList.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("NodeList"))
+	if err := cl.List(ctx, nodeList); err != nil {
+		return false, err
+	}
+	nodesPerZone := make([]int, len(zones))
+	zonesIncomplete := len(zones)
+forNode:
+	for _, node := range nodeList.Items {
+		nodeZone := node.Labels[corev1.LabelTopologyZone]
+		for i, zone := range zones {
+			if strings.HasSuffix(nodeZone, zone) {
```

Comment:
Can you add a comment here about why `strings.HasSuffix` is used and that all of this is rather a best effort approach, i.e. does not work for all providers which have zone name divergence?

Reply:
Added comment in https://github.com/gardener/gardener/pull/8970/commits/bb0f380dbdb71823b8e2c11b86c0baf70ead763d.